synapse/models/person.py

```diff
@@ -137,47 +190,19 @@ def getBaseModels():
                     # FIXME add an optional bounding box
                 )),
 
-                ('ps:hasuser', {'ptype': 'ps:hasuser'}, (
-                    ('person', {'ptype': 'ps:person'}),
-                    ('user', {'ptype': 'inet:user'}),
-                    ('seen:min', {'ptype': 'time:min'}),
-                    ('seen:max', {'ptype': 'time:max'}),
-                )),
-
-                ('ps:hasalias', {'ptype': 'ps:hasalias'}, (
-                    ('person', {'ptype': 'ps:person'}),
-                    ('alias', {'ptype': 'ps:name'}),
-                    ('seen:min', {'ptype': 'time:min'}),
-                    ('seen:max', {'ptype': 'time:max'}),
-                )),
-
-                ('ps:hashost', {'ptype': 'ps:hashost'}, (
-                    ('person', {'ptype': 'ps:person'}),
-                    ('host', {'ptype': 'it:host'}),
-                    ('seen:min', {'ptype': 'time:min'}),
-                    ('seen:max', {'ptype': 'time:max'}),
-                )),
-
-                ('ps:hasphone', {'ptype': 'ps:hasphone'}, (
-                    ('person', {'ptype': 'ps:person'}),
-                    ('phone', {'ptype': 'tel:phone'}),
-                    ('seen:min', {'ptype': 'time:min'}),
-                    ('seen:max', {'ptype': 'time:max'}),
-                )),
-
-                ('ps:hasemail', {'ptype': 'ps:hasemail'}, (
-                    ('person', {'ptype': 'ps:person'}),
-                    ('email', {'ptype': 'inet:email'}),
+                ('ps:has', {}, [
+                    ('person', {'ptype': 'ps:person', 'ro': 1, 'req': 1,
+                        'doc': 'The person that has the given node.'}),
+                    ('xref', {'ptype': 'propvalu', 'ro': 1, 'req': 1,
+                        'doc': 'The prop=valu that is referenced as part of the FIXME.'}),
+                    ('xref:node', {'ptype': 'ndef', 'ro': 1, 'req': 1,
+                        'doc': 'FIXME.'}),
+                    ('xref:prop', {'ptype': 'str', 'ro': 1,
+                        'doc': 'The property (form) of the referenced object, as specified by the propvalu.'}),
```

Comment:
Suggested: "The property (form) of the object or resource that is owned or controlled by the person."